pqlite/storage/table.py

```diff
@@ -211,16 +213,17 @@ def insert(
             values.append(doc_value)
             docs_size += 1
 
-        cursor = self._conn.cursor()
-        if docs_size > 1:
-            cursor.executemany(sql, values[:-1])
+        with self._conn_lock:
```

Comment:
why do u need a lock here, who is starting a thread?

Reply:
This change is made for executor with Jina 3.x. In 3.x, the executor runtime changed to GRPC runtime. I just observed that the sqlite connection instance `self._conn` created at `__init__` is used in different threads.  Thus, I add a lock here. 

But this is only for debugging and is not the final commit. Maybe this lock will be removed.  

```
ProgrammingError('SQLite objects created in a thread can only be used in that same thread. The object was created in thread id 8674164224 and this is thread id 120472346624
```